### PR TITLE
 Fix print pf.type.field on types above 250 bytes

### PR DESCRIPTION
**Pull request by morbith-dqtz**

### CODE REVIEW

**Comment by radare** [comment]
Please never merge in branches that you are PRs. otherwise the commit history and changes get messed up with other stuff that is unrealtd to your commit. always use cherry-pick and rebase-i to squash them

File: libr/core/cmd_print.c [hunk 1]
```diff
@@ -1031,6 +1031,11 @@ static void cmd_print_format(RCore *core, const char *_input, const ut8* block,
 			/* Load format from name into fmt to get the size */
 			/* This make sure the whole structure will be printed */
 			const char *fmt = NULL;
+			if (strchr (name, '.')) {
+				char *delim = strchr (name, '.');
```
Comment (radare) [inline libr/core/cmd_print.c]: if delim is null it will crash

File: libr/core/cmd_print.c [hunk 2]
```diff
@@ -1031,6 +1031,11 @@ static void cmd_print_format(RCore *core, const char *_input, const ut8* block,
...
+				int len = strlen (name) - strlen (delim);
+				name = strndup (name, len);
```
Comment (radare) [inline libr/core/cmd_print.c]: use r_str_ndup(). strndup is not portable

File: libr/core/cmd_print.c [hunk 3]
```diff
@@ -1031,6 +1031,11 @@ static void cmd_print_format(RCore *core, const char *_input, const ut8* block,
 			/* Load format from name into fmt to get the size */
 			/* This make sure the whole structure will be printed */
 			const char *fmt = NULL;
+			if (strchr (name, '.')) {
+				char *delim = strchr (name, '.');
+				int len = strlen (name) - strlen (delim);
```
Comment (radare) [inline libr/core/cmd_print.c]: no need to strlen, just substract pointers.

File: libr/core/cmd_print.c (hunk 2 above)
Comment (radare) [reply libr/core/cmd_print.c]: check if name  is null to avoid further issues

Comment (radare) [reply libr/core/cmd_print.c]: free name in a temporary variable. this line is leaking

**Comment by radare** [comment]
why closed?

File: libr/core/cmd_print.c [hunk 4]
```diff
@@ -1031,6 +1031,17 @@ static void cmd_print_format(RCore *core, const char *_input, const ut8* block,
 			/* Load format from name into fmt to get the size */
 			/* This make sure the whole structure will be printed */
 			const char *fmt = NULL;
+			if (strchr (name, '.')) {
+				char *delim = strchr (name, '.');
+				if (delim) {
+					int len = delim - name;
+					char *name_c = r_str_ndup (name, len);
```
[the remark this answers is not in the document]
Comment (morbith-dqtz) [reply libr/core/cmd_print.c]: Wooo my eyes are bleeding, don't ask xD

**Comment by morbith-dqtz** [comment]
This way is segfaulting in some cases. 
I have a better option tested and seems pretty stable.

I close this one and try to to things from a branch to not mess like this again.

**Pull request closed without merging**